_posts/python/2016-07-26-get-slope-aspect-from-digital-elevation-model.md

```diff
@@ -22,162 +22,83 @@ With this data, we are able to use programs in order to analyze certain physical
-           [        nan,         nan,         nan, ...,  0.32364485,
-             0.44950795,  0.41471453],
-           [        nan,         nan,  0.04311538, ...,  0.43404787,
-             0.4197514 ,  0.28500046],
-           ..., 
-           [ 0.44798625,  0.44798625,  0.51895544, ...,  0.1062964 ,
-             0.12971104,  0.12971104],
-           [ 0.3983211 ,  0.47653369,  0.52412218, ...,  0.07775003,
-             0.10756881,  0.1297108 ],
-           [ 0.42918458,  0.5150215 ,  0.55793996, ...,  0.07775003,
-             0.10756881,  0.08583692]])
-
-
-
-
 ```python
-aspect
+plt.imshow(shasta_dem, interpolation='none')
+plt.colorbar()
+plt.show()
 ```
 
 
+![png](/images/get-slope-aspect-from-digital-elevation-model_files/get-slope-aspect-from-digital-elevation-model_7_0.png)
 
 
-    array([[        nan,         nan,         nan, ...,  2.49694673,
-             2.6124799 ,  3.14159265],
-           [        nan,         nan,         nan, ...,  2.49694673,
-             2.6124799 ,  2.24437453],
-           [        nan,         nan,  3.14159265, ...,  2.67698894,
-             2.37318076,  2.22072197],
-           ..., 
-           [ 0.53113069,  0.53113069,  0.59700458, ...,  4.29676177,
-             3.98922813,  3.98922813],
-           [ 0.2466342 ,  6.14671481,  6.09656191, ...,  4.12762132,
-             3.78846596,  3.98923025],
-           [ 0.        ,  0.        ,  0.        , ...,  4.12762132,
-             3.78846596,  3.14159265]])
-
+## Compute and plot slope and aspect
 
+We can use the `rd.TerrainAttribute` function to compute slope and aspect for each pixel. Note that there are multiple ways to represent the slope values. See https://richdem.readthedocs.io/en/latest/python_api.html#richdem.TerrainAttribute for more options.
 
-Next, let's visualize aspect and slope using the `imshow` function. 
+To visualize slope and aspect, we can use the `rdShow` function. 
 
 
 ```python
-# use a log scale to normalize the scale a bit
-plt.imshow(np.log(mag), cmap="magma")
+slope = rd.TerrainAttribute(shasta_dem, attrib='slope_riserun')
+rd.rdShow(slope, axes=False, cmap='magma', figsize=(8, 5.5))
+plt.show()
 ```
 
 
-
-
-    <matplotlib.image.AxesImage at 0x111dab150>
-
-
-
-
-![png](/images/get-slope-aspect-from-digital-elevation-model_files/get-slope-aspect-from-digital-elevation-model_10_1.png)
+![png](/images/get-slope-aspect-from-digital-elevation-model_files/get-slope-aspect-from-digital-elevation-model_9_0.png)
 
 
 
 ```python
-plt.imshow(aspect, cmap = "viridis")
+aspect = rd.TerrainAttribute(shasta_dem, attrib='aspect')
+rd.rdShow(aspect, axes=False, cmap='jet', figsize=(8, 5.5))
+plt.show()
 ```
 
 
-
-
-    <matplotlib.image.AxesImage at 0x113efe110>
-
-
-
-
-![png](/images/get-slope-aspect-from-digital-elevation-model_files/get-slope-aspect-from-digital-elevation-model_11_1.png)
+![png](/images/get-slope-aspect-from-digital-elevation-model_files/get-slope-aspect-from-digital-elevation-model_10_0.png)
```

Comment:
We can update this in our future build. i've created a notebook template that will allow us to populate a proper alt tag rather than "png". It's nice because it will add a caption to the image too and then thepage is more accessible to those with screen readers... let's not forget this when we ingest your tutorials into the build!

Reply:
Cool - I'm curious to see how you did that. Sounds like we will need to do that for all of these old tutorials at some point. Did you want to do it for this tutorial now, or later on in bulk?

Reply:
oh! i can't tell you how happy i was to find the template stuff. Here is the template? Essentially i  make a call to the "caption" tag which you have to add to the metadatata with a value (the actual caption text. It's just json in the notebook. you can look at the json  too...

Here is the call to nbconvert - this is where the template is specified.
https://github.com/earthlab/earth-analytics-python/blob/master/scripts/generate_posts.py#L157
here is the template:

https://github.com/earthlab/earth-analytics-python/blob/master/scripts/jupyter_templates/jekyll-markdown.tpl

And look here for the caption tag in the json (just search for "caption" on the page as gh isn't allowing me to view json with line numbers. 

https://raw.githubusercontent.com/earthlab/earth-analytics-python/master/notebooks/final-notebooks/courses/earth-analytics-python/08-multispectral-remote-sensing-fire/clouds-cloud-masks/2017-03-01-fire02-replace-na-values-in-raster-with-different-raster.ipynb?token=AHS3qg30bEQQIj7qtKrlQbYtE-XQsgtaks5b_BSLwA%3D%3D

I think its probably worth doing it when we combine out builds. gina can work on it. that could be one of the first things we do - merge tutorials with lessons into one repo and adjust the script so that it handles both. i wanted to modify the script anyway to do several things.

Anyway prob not worth doing now :)